enhancements/etcd/etcd-size-tuning.md

```diff
@@ -0,0 +1,192 @@
+
+### Goals
+
+* Add a configuration option using resource.Quantity to allow for human readable values; i.e. 16GiB.
+* Remove the parameters from the podspec rendering and replace with the value.
+* Add an API to allow admins to change the value.
+* Add blocker to disallow shrinking the size once set to a larger value.
+
+### Non-Goals
+
+* Handle consuming value changes without an etcd rollout.
+
+## Proposal
+
+Allow the customer/admin access to set the backend quota directly through a human readable value. The only restriction is that they are disallowed from setting the value lower than the current.
+
+The default value or unset value would be the current, hardcoded backend quota: 8GiB. This would allow for upgrades where this configuration value wouldn't nesscessarily be set.
+
+This would allow for more flexibility, but would also likely increase the number of support cases as the memory requirements will be higher: changing quota from 8GiB to 16GiB is not double the memory, it'll be 6x the memory since each etcd member could take up to 16GiB.
+We will need to work with the perf team to understand the memory requirements of largers quotas, and the effect of the quota on things like defragmentation.
+A downside is that with the profiles, discussed in the Alternatives section, there are a discrete, small, number of permutations to test, giving more confidence of the exact effects a given profile has on performance/memory requirements.
+This is a minor consideration as setting the quota to a arbitrarily higher value does not mean that etcd will take up that much space; it will depend on the workload.
+
+### Workflow Description
+
+1. The cluster administrator decides to change the etcd backend quota from the default (8GiB) to a larger value; e.g. 20GiB.
+2. They set the new value in the API server.
+3. An etcd redeployment will be automatically run which restarts the etcd pods which consume the new value.
+
+#### Variation [optional]
+
+1. The cluster administrator wants to set the quota from a larger value (e.g. 32GiB) to a smaller one (8GiB).
+2. They attempt to set the new value in the API server.
+3. The API server should return an error notifying them that the value is invalid - not allowed to shrink the quota.
+
+### API Extensions
+
+In operator/v1/0000_12_etcd-operator_01_config-TechPreviewNoUpgrade.crd.yaml:
+- Add BackendSizeQuota of type *resource.Quantity
+  - With default '8GiB'.
+  - With validation disallow a decrease in value.
+
+### Implementation Details/Notes/Constraints [optional]
+
+None
+
+### Risks and Mitigations
+
+* We will need to test the larger size to ensure that everything works as expected; especially defragmentation.
+* Defragmentation, and its effects, will need to be tested as the API server will be unresponsive during this time; we will need to understand the general magnitude of defrag time and documentation those expectations.
```

Comment:
This will impact API Server SLOs. Having a bigger size, greater defragmentation == less potential availability for the apiserver. SLA implications for managed services.

Reply:
managed services do not defrag via the etcd operator. I also just want to leave a quick note that etcd still has two functional other members to reach while CEO defragments one, so I doubt this causes any availability issues.

@dusk125 can you update the scope/non-goals to not apply to managed services please? And maybe defuse this section, the etcd member under defrag will not be able to serve any writes, that doesn't mean the apiserver is unavailable.

Reply:
etcd API would be where my assumption was wrong